pkg/scheduler/scheduler.go

```diff
@@ -471,9 +471,12 @@ func (s *Scheduler) getAssignments(log logr.Logger, wl *workload.Info, snap *cac
 	if wl.CanBePartiallyAdmitted() {
 		reducer := flavorassigner.NewPodSetReducer(wl.Obj.Spec.PodSets, func(nextCounts []int32) (*partialAssignment, bool) {
 			assignment := flvAssigner.Assign(log, nextCounts)
-			if assignment.RepresentativeMode() == flavorassigner.Fit {
+			mode := assignment.RepresentativeMode()
+			if mode == flavorassigner.Fit {
 				return &partialAssignment{assignment: assignment}, true
-			} else if assignment.RepresentativeMode() == flavorassigner.Preempt {
+			}
+
+			if mode == flavorassigner.Preempt {
```

Comment:
> [This test](https://github.com/kubernetes-sigs/kueue/blob/00111d9aada59e1a6d8d90b9847cf7451bc5be76/pkg/scheduler/scheduler_test.go#L1194) exits [here](https://github.com/kubernetes-sigs/kueue/blob/00111d9aada59e1a6d8d90b9847cf7451bc5be76/pkg/scheduler/scheduler.go#L467-L469), as we find a preemption target above. The PartialAdmission preemption path is not tested.

after deletion of this preemption block, the tests noted in #2799 still passes:
```
go test ./pkg/scheduler -run "TestSchedule/partial_admission_single_variable_pod_set,_preempt_first"
```

Reply:
That test checks that preemption is done and wl admitted in full (without checking if it fits partially and decreasing the count) if that preemption is an option.